### PR TITLE
#7613: adds required methods to strict messenger methods for toolbar icon work

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Part of #7613 

## Reviewer Tips

- See  `src/background/messenger/strict/*`
- Required for changes in https://github.com/pixiebrix/pixiebrix-extension/pull/7632

## Checklist

- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer @mnholtz

### CODE REVIEW

**Review by mnholtz — APPROVED** [review]

File: src/registry/memoryRegistry.ts [hunk 1]
```diff
@@ -282,7 +282,7 @@ class MemoryRegistry<
 
     const remoteItems: Item[] = [];
     for (const raw of packages) {
-      const item = this.parse(raw.config);
+      const item = this.parse(raw?.config);
```
Comment (twschiller) [inline src/registry/memoryRegistry.ts:285]: I don't think this is correct, it should be `!` to assert the package is non-null. Under what conditions can getByKinds return null elements?

File: src/registry/packageRegistry.ts [hunk 2]
```diff
@@ -224,28 +229,34 @@ async function replaceAll(packages: PackageVersion[]): Promise<void> {
...
+    rawConfig: null,
   };
 }
 
 /**
  * Return the latest version of a brick, or null if it's not found.
  * @param id the registry id
  */
-export async function find(id: string): Promise<PackageVersion | null> {
+export async function find(id: string): Promise<Nullishable<PackageVersion>> {
```
Comment (twschiller) [inline src/registry/packageRegistry.ts:259]: 👍

File: src/registry/packageRegistry.ts [hunk 3]
```diff
@@ -107,10 +108,12 @@ async function openRegistryDB() {
   return database;
 }
 
-function latestVersion(versions: PackageVersion[]): PackageVersion | null {
+function latestVersion(
```
Comment (twschiller) [inline src/registry/packageRegistry.ts:111]: Why is `Nullishable<PackageVersion>` required? I don't think `await db.getAllFromIndex` would return null elements

File: src/registry/memoryRegistry.ts (hunk 1 above)
Comment (grahamlangford) [reply src/registry/memoryRegistry.ts:285]: I don't know. I was doing the bare minimum to make TypeScript happy. I can change it if we'd  rather it throw an error if it was nullish.

**Review by fungairino — APPROVED** [review]

File: src/registry/packageRegistry.ts (hunk 3 above)
Comment (grahamlangford) [reply src/registry/packageRegistry.ts:111]: It seemed to be required at one point. But you're right, it's not. Fixed.

**Pull request merged**